### PR TITLE
Chapter 9: Advanced login

### PR DESCRIPTION
**Pull request by tuanvuP**
Screenshot from 2019-06-19 17-04-57

### CODE REVIEW

**Comment by tuanvuP** [comment]
ready

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -4,15 +4,36 @@ def log_in user
...
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def forget user
+    user.forget
+    cookies.delete :user_id
+    cookies.delete :remember_token
```
Comment (VuThiTranVan) [inline app/helpers/sessions_helper.rb]: viết dồn 1 line dc nhé gái
`cookies.delete [:user_id, :remember_token]

File: app/models/user.rb [hunk 2]
```diff
@@ -13,13 +15,33 @@ class User < ApplicationRecord
...
+    end
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.nil?
```
Comment (VuThiTranVan) [inline app/models/user.rb]: `if remember_digest.nil?` thành `unless remember_digest`

**Comment by VuThiTranVan** [comment]
merged
fix comment rồi mợt hì

**Pull request merged**